Added unit tests for JSON module

DobromirM opened the pull request:
Added unit tests for some of the edge cases that were not being tested.
(Line 59, 60 and 66 from the JsonStructureWriter class.)

brohitbrose commented:
Looks good! Just a couple of things:

- I believe that the `imports` will trigger a checkstyle warning, in particular the space between the groups and the use of a `.*` import. The checkstyle definition can be found under the `swim` repository in `swim-system-java/checkstyle.xml`. To verify, you can run `gradle checkstyleMain` and `gradle checkstyleTest` from the `swim-system-java` directory.

- This one's our fault: we've decided to keep all the development in the [`swim` repository itself](https://github.com/swimos/swim), which is a superset of many other repos including `swim-core-java`. These "subrepos" are now going to be "read-only" views of sections of the `swim` repo. Sorry about that. We'll disable PRs on these subrepos ASAP.

What you can do is apply these fixes in the `swim` repository instead, and open a new PR there. Thank you for your interest!

DobromirM commented:
Moved the PR to the correct repository.

I also changed the import to match the style, but `gradle checkstyleMain` and `gradle checkstyleTest` did not show a warning when I ran them with `.*` import. Might need to add a TODO / open a bug to fix it.
